firedrake/slate/slac/compiler.py

```diff
@@ -224,7 +229,23 @@ def generate_kernel_ast(builder, statements, declared_temps):
     # Now we wrap up the kernel ast as a PyOP2 kernel and include the
     # Eigen header files
     include_dirs = builder.include_dirs
-    include_dirs.extend(["%s/include/eigen3/" % d for d in PETSC_DIR])
+
+    def extract_eigen_include_dir(PETSC_ARCH):
+        try:
+            file = open("%s/lib/petsc/conf/petscvariables" % (PETSC_ARCH))
```

Comment:
Assumes current working directory is `PETSC_DIR`, I think.

Reply:
No, pyOP2 puts $PETSC_DIR+PETSC_ARCH in `PETSC_ARCH`. Don't ask me why...
Does it even make sense to use pyop2 for that at all ?